cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectDockerfile.java

```diff
@@ -284,6 +313,61 @@ private void addZipArtifact(PrintWriter writer, String connectorPath, ZipArtifac
         writer.println();
     }
 
+    /**
+     * Add command sequence for downloading Maven artifact
+     *
+     * @param writer            Writer for printing the Docker commands
+     * @param connectorPath     Path where the connector to which this artifact belongs should be downloaded
+     * @param mvn               The maven artifact
+     */
+    private void addMavenArtifact(PrintWriter prefixWriter, PrintWriter writer, String connectorPath, MavenArtifact mvn) {
+        validateGavPresence(mvn);
+        String repo = mvn.getRepository() == null ? MavenArtifact.DEFAULT_REPOSITORY : maybePatchRepository(mvn.getRepository());
+        String artifactHash = Util.sha1Prefix(mvn.getGroup() + mvn.getArtifact() + mvn.getVersion());
+        String artifactDir = connectorPath + "/" + artifactHash;
+
+        String downloadPomCmd =  String.format("curl -L --output pom.xml %s%s/%s/%s/%s-%s.pom",
+                repo,
+                mvn.getGroup().replace(".", "/"), //org.apache.camel is translated as org/apache/camel in the URL
+                mvn.getArtifact().replace(".", "/"),
+                mvn.getVersion(),
+                mvn.getArtifact(),
+                mvn.getVersion());
+
+        String downloadJarCmd = String.format("curl -L --create-dirs --output " + artifactDir + "/%s-%s.jar %s%s/%s/%s/%s-%s.jar",
+                mvn.getArtifact(),
+                mvn.getVersion(),
+                repo,
+                mvn.getGroup().replace(".", "/"), //org.apache.camel is translated as org/apache/camel in the URL
+                mvn.getArtifact().replace(".", "/"),
+                mvn.getVersion(),
+                mvn.getArtifact(),
+                mvn.getVersion());
+
+        prefixWriter.println("FROM maven:3.8.1-openjdk-17-slim AS download" + artifactHash);
```

Comment:
* I'm not sure we should use Java 17 for this. Can't there be different artifacts for Java 17 in some cases? Does that impact what is downloaded?
* Since we use UBI8 as base image, I wonder if we should use https://catalog.redhat.com/software/containers/ubi8/openjdk-11/5dd6a4b45a13461646f677f4?container-tabs=overview instead
* I wonder if we should repackage this image similarly how we do it for Kaniko.